modules/rostests/apitests/atl/CAtlArray.cpp

```diff
@@ -6,6 +6,11 @@
  *              Copyright 2019 Mark Jansen (mark.jansen@reactos.org)
  */
 
+#ifdef __GNUC__
+// this is an intended behaviour
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
```

Comment:
I am not sure if this restores at the end of the file automatically?
If not, this might need to be restored after the problematic section

Reply:
```
In file included from C:/rosgit/modules/rostests/apitests/atl/CAtlArray.cpp:21:
C:/rosgit/sdk/lib/atl/atlcoll.h: In instantiation of 'static void ATL::CElementTraitsBase<T>::RelocateElements(T*, T*, size_t) [with T = CCreature; size_t = unsigned int]':
C:/rosgit/sdk/lib/atl/atlcoll.h:201:38:   required from 'bool ATL::CAtlArray<E, ETraits>::GrowAllocatedData(size_t) [with E = CCreature; ETraits = ATL::CElementTraits<CCreature>; size_t = unsigned int]'
C:/rosgit/sdk/lib/atl/atlcoll.h:334:14:   required from 'bool ATL::CAtlArray<E, ETraits>::SetCount(size_t, int) [with E = CCreature; ETraits = ATL::CElementTraits<CCreature>; size_t = unsigned int]'
C:/rosgit/modules/rostests/apitests/atl/CAtlArray.cpp:75:26:   required from here
C:/rosgit/sdk/lib/atl/atlcoll.h:102:16: error: 'void* memmove(void*, const void*, size_t)' writing to an object of type 'struct CCreature' with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
         memmove(Dest, Source, NumElements * sizeof(T));
         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/rosgit/modules/rostests/apitests/atl/CAtlArray.cpp:23:8: note: 'struct CCreature' declared here
 struct CCreature
        ^~~~~~~~~
```